File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = [['#'] * n for _ in range(n)]
    
    idx = 0
    for a, b in zip(arr1, arr2): 
        line = bin(a | b)[2:].zfill(n)      # 2진수 변환 후, OR 연산자, n만큼 0채우기
        for i in range(len(line)):
            if line[i] == '0':              # 0일 때는 공백
                answer[idx][i] = ' '
        idx += 1
    
    
    return [''.join(a) for a in answer]

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
function solution(n, arr1, arr2) {
    var answer = [[]];
    for (let i = 0; i < n; i++) {
        answer[i] = []
        var bin_2 = (arr1[i] | arr2[i]).toString(2).padStart(n, 0)
        for (let j = 0; j < bin_2.length; j++) {
            if (bin_2[j] === '1') answer[i].push('#')
            else answer[i].push(' ')
        }
    }
    answer = answer.map(e => e.join(''))
    return answer;
}